src/collector/Metrics.cpp

```diff
@@ -43,7 +43,7 @@ void Distribution::add_sample(uint64_t sample)
 
     uint64_t min_diff = m_bins[1].value - m_bins[0].value;
     int idx_min = 0;
-    for (int i = 1; i < (m_bins.size() - 1); ++i) {
+    for (size_t i = 1; i < (m_bins.size() - 1); ++i) {
         uint64_t cur_diff = m_bins[i].value - m_bins[i + 1].value;
```

Comment:
Why does it change order of subtraction? When you calculates `min_diff` you subtract value of the first element from value of the second element. When you calculates `cur_diff` you subtract value of the next element from value of the previous element.